rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -158,16 +143,48 @@ pub mod checker {
         };
         serde_json::to_vec(&r).map_or(null(), vec_to_c_char)
     }
+
+    unsafe fn apply_block_inner(id: u64, block_trace: *const c_char) -> Result<RowUsage, Error> {
+        log::debug!(
+            "ccc apply_block raw input, id: {:?}, block_trace: {:?}",
+            id,
+            c_char_to_str(block_trace)?
+        );
+        let block_trace = c_char_to_vec(block_trace);
+        let traces = serde_json::from_slice::<BlockTrace>(&block_trace)?;
+
+        let r = panic::catch_unwind(|| {
+            CHECKERS
+                .get_mut()
+                .ok_or(anyhow!(
+                    "fail to get circuit capacity checkers map in apply_block"
+                ))?
+                .get_mut(&id)
+                .ok_or(anyhow!(
+                    "fail to get circuit capacity checker (id: {id:?}) in apply_block"
+                ))?
+                .estimate_circuit_capacity(&[traces])
+        });
+        match r {
+            Ok(result) => result,
+            Err(_) => {
+                return Err(anyhow!(
+                    "estimate_circuit_capacity (id: {id:?}) panics in apply_block"
```

Comment:
```suggestion
                    "estimate_circuit_capacity (id: {id:?}) error in apply_block"
```

Reply:
when `ccc` panics, it goes here. So I think `panics` will be better.